Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -564,6 +564,37 @@ public NodejsProjectNode(NodejsProjectPackage package)
                 // scan for files which were loaded from cached analysis but no longer
                 // exist and remove them.
                 _analyzer.ReloadComplete();
+
+                Task.Factory.StartNew(LogProjectInformation);
+            }
+        }
+
+        /// <summary>
+        /// Log generic characteristics about the project, including number of files of each type in the project.
+        /// </summary>
+        private void LogProjectInformation() {
+            var fileTypeInfo = new Dictionary<string, int>();
+            foreach (var node in this.DiskNodes) {
```

Comment:
Thinking about this more, I'm not confident that iterating here is safe. If the dictionary is changed on another thread, the iterator will be invalidated. I haven't been able to repo that in my testing, but it seems like it should be possible to trigger an exception from that.

Even moving this logic to the main thread may not be safe since it's not clear to me if any background threads could end up updating `DiskNodes`.